k8s/models/secret.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+# Copyright 2017-2022 The FIAAS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import six
+
+from .common import ObjectMeta
+from ..base import Model
+from ..fields import Field
+
+
+class Secret(Model):
+    class Meta:
+        url_template = "/api/v1/namespaces/{namespace}/secrets/{name}"
+
+    metadata = Field(ObjectMeta)
+    data = Field(dict)
```

Comment:
I wonder if it would be useful to also support the `stringData` field?
Preferably with a new field type to indicate that this is a write-only field. On the other hand, I don't think we have that anywhere else, so maybe it's not needed. Also, I'm not sure when the `stringData` field was introduced. Might cause problems if expecting to use it in a cluster that doesn't support it?

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#secret-v1-core

Reply:
Good question! The only use for this field is to provide non binary data, which will end up being merged into `data` field, so it's kind of redundant. Unless someone has a specific use case for this, I'd say it's not needed.

Reply:
Seems that we do not have any cluster without the stringData field so they will be no problem for add it.

Reply:
@mortenlj Added field stringData. Please give it a look, I've created a new WriteOnlyField type, I think is what you expected.

Also, is there a way to get rid of those picky errors about the methods defined not as functions? I think is the proper way to do it and follows the same structure as the rest of the models.

Reply:
> Added field stringData. Please give it a look, I've created a new WriteOnlyField type, I think is what you expected.

:+1: 

> Also, is there a way to get rid of those picky errors about the methods defined not as functions? I think is the proper way to do it and follows the same structure as the rest of the models.

Technically, they could be moved to the top-level, outside the containing class, since it's not actually using anything from the class. Personally, I prefer to group tests in a class, but it's probably not necessary when there's only one class in the test file.

I guess you could decorate them with `@staticmethod` and see if that 1) works, 2) silences the linter. Or just move them to the top-level. Or just ignore them. :shrug: